Sources/_StringProcessing/ByteCodeGen.swift

```diff
@@ -293,13 +293,15 @@ extension Compiler.ByteCodeGen {
   mutating func emitMatcher(
     _ matcher: @escaping _MatcherInterface,
     into capture: CaptureRegister? = nil
-  ) {
+  ) throws {
```

Comment:
Hmm why does the compiler need to throw? It's only the underlying matcher that throws, right?

Reply:
Hmm you're right that it doesn't. I didn't understand how this works well so I thought it was needed. For my information why does the compiler need to throw in the case like `emitTransform` and `emitCharacter`?

Reply:
I think it's because they call `emitNode` which throws. `emitNode` throws because `emitReference` throws. `emitReference` throws because it needs to check the validity of references and throw `RegexCompilationError`.